### PR TITLE
Deprecate DisableTLS field, add noVerifyTLS in email.smtp config

### PR DESCRIPTION
**Pull request by arussellsaw**
This PR fixes https://github.com/sourcegraph/sourcegraph/issues/13043

❓  There also seems to be typescript files generated from the json schema, but i couldn't work out how to generate them, does that need doing in this PR?

todo:
- [x] add changelog entry

### CODE REVIEW

**Comment by arussellsaw** [comment]
i'm getting a build failure because of codecov but i'm not sure why... perhaps this file has too low test coverage?

**Review by tsenart — APPROVED** [review]
Ignore the Codecov failure here, I don't think it's important in this case.

**Comment by tsenart** [comment]
> There also seems to be typescript files generated from the json schema, but i couldn't work out how to generate them, does that need doing in this PR?

Did you run `./dev/generate.sh`?

**Comment by arussellsaw** [comment]
i've run ./dev/generate.sh but that didn't update any of the typescript stuff. i discovered another command `yarn generate` but that doesn't yield any diff

File: internal/txemail/txemail.go [hunk 1]
```diff
@@ -117,7 +117,8 @@ func Send(ctx context.Context, message Message) error {
 		return fmt.Errorf("invalid SMTP authentication type %q", conf.EmailSmtp.Authentication)
 	}
 
-	if conf.EmailSmtp.DisableTLS {
+	// TODO: 2020/11/12 @arussellsaw, after next release delete DisableTLS option
```
Comment (unknwon) [inline internal/txemail/txemail.go:120]: It is best to create an issue to track this so we don't forget :)

**Pull request merged**